### PR TITLE
feat: 🎸 blog section created

### PR DESCRIPTION
**Pull request by abrahamcalsin**

### CODE REVIEW

File: _articles/building-a-responsible-aI-solution-principles-into-practice.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+This is part 2 of my series of posts on fairness in AI, reflections from a participant in the Monetary Authority of Singapore's (MAS) Global Veritas Challenge. In my first post, I discussed the need to adopt a human-centric approach when dealing with issues of fairness in AI systems. After hours of brainstorming, user interviews and smashing bits of code together, we created VerifyML, an open-source framework to help organisations build fair and responsible models. The solution was one of the three winning submissions of the Veritas challenge!
+
+My teammates have written about VerifyML in detail, so if you are interested in giving it a spin, check out the following introductory posts:
+
+- Introducing VerifyML
+- A Quickstart Guide to VerifyML Part 1
+- A Quickstart Guide to VerifyML Part 2
+- Comparing VerifyML, AI Fairness 360 and Fairlearn
+
+##
```
Comment (yonycalsin) [inline _articles/building-a-responsible-aI-solution-principles-into-practice.md:18]: remove this unused line

File: _articles/why-i-build-design-systems-with-stitches-and-radix.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+I've lost count of how many times I've been asked to build complex components from scratch—dialogs, tooltips, dropdown menus, popovers, sheets, tabs.
+
+But they were all flawed and lacked fundamental accessibility features. When working against a deadline, it's almost impossible to build them correctly.
+
+So for the last couple of years, we at Modulz worked hard to facilitate how teams can build, maintain and scale their design systems—all whilst adhering to the WAI-ARIA design pattern.
+
+If you've never heard of Stitches or Radix Primitives before, here's a little background.
+
+Stitches
+Stitches is a styling solution focusing on component architecture and developer experience.
```
Comment (yonycalsin) [inline _articles/why-i-build-design-systems-with-stitches-and-radix.md:20]: Why are you adding the word "Stitches" twice?

File: lib/api.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+export function getArticleSlugs() {
+  return fs.readdirSync(articlesDirectory)
+}
+
+export function getArticleBySlug(slug: string, fields: string[] = []) {
+  const realSlug = slug.replace(/\.md$/, '')
+  const fullPath = join(articlesDirectory, `${realSlug}.md`)
+  const fileContent = fs.readFileSync(fullPath, 'utf8')
+  const { data, content } = matter(fileContent)
```
Comment (yonycalsin) [inline lib/api.ts:15]: pls, use space among these lines

File: lib/api.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import fs from 'fs'
+import { join } from 'path'
+import matter from 'gray-matter'
+
+const articlesDirectory = join(process.cwd(), '_articles')
```
Comment (yonycalsin) [inline lib/api.ts:5]: I do not like the name of this folder, consider, rename to `content/blog`.

File: pages/articles/[slug].tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+export default Article
+
+type ParamsProps = {
+  params: {
+    slug: string
+  }
+}
+
+export async function getStaticProps({ params }: ParamsProps) {
```
Comment (yonycalsin) [inline pages/articles/[slug].tsx:54]: i think that you can use `GetStaticPropsContext` type from `next` package

File: src/components/article-body/index.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// import markdownStyles from './markdown-styles.module.css'
```
Comment (yonycalsin) [inline src/components/article-body/index.tsx:1]: Remove this unnecessary comment line

**Review by yonycalsin — CHANGES_REQUESTED** [review]

File: lib/api.ts (hunk 4 above)
Comment (abrahamcalsin) [reply lib/api.ts:5]: that folder also contains the code for google analitycs, is it considerable to change the folder name to content/blog, or have I misunderstood?

Comment (abrahamcalsin) [reply lib/api.ts:5]: or do you mean the name of the `articlesDirectory` constant?

Comment (yonycalsin) [reply lib/api.ts:5]: the idea of the `content` folder is to store content and not code.

```ts
content/blog/*.md
content/snippets/*.md
content/books/*.md
```

so your google-analytics code should be in the src/_document.tsx folder.

Comment (abrahamcalsin) [reply lib/api.ts:5]: so you mean copy the code from the lib/ga/index.tsx file, into the _document.tsx file?

copy-paste

Comment (yonycalsin) [reply lib/api.ts:5]: > or do you mean the name of the articlesDirectory constant?
@abrahamcalsin  yep

Comment (yonycalsin) [reply lib/api.ts:5]: >so you mean copy the code from the lib/ga/index.tsx file, into the _document.tsx file?

nope

Comment (abrahamcalsin) [reply lib/api.ts:5]: > 
> @abrahamcalsin yep

Ok, I understand

**Review by yonycalsin — APPROVED** [review]

**Comment by yonycalsin** [comment]
Good Job 🎉

**Pull request merged**